modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlElement.java

```diff
@@ -0,0 +1,391 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.io;
+
+import static com.opengamma.strata.collect.Guavate.ensureOnlyOne;
+import static com.opengamma.strata.collect.Guavate.toImmutableList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.xml.namespace.QName;
+
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.MetaBean;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.Messages;
+
+/**
+ * A single element in the tree structure of XML.
+ * <p>
+ * This class is a minimal, lightweight representation of an element in the XML tree.
+ * The element has a name, attributes, and either content or children.
+ * <p>
+ * Note that this representation does not express all XML features.
+ * No support is provided for processing instructions, comments or mixed content.
+ * In addition, it is not possible to determine the difference between empty content and no children.
+ * <p>
+ * There is no explicit support for namespaces.
+ * When creating instances, the caller may choose to use a convention to represent namespaces.
+ * For example, element and attribute names may use prefixes or the standard {@link QName} format.
+ */
+public final class XmlElement
```

Comment:
The JDK DOM API is a direct port of the standard DOM. Its a horrible API, very difficult to create and query. It also supports more than is really necessary. As shown here, one class is all that is really needed.